### PR TITLE
feat(controller): support realtime log

### PR DESCRIPTION
**Pull request by jialeicui**
## Description
Support realtime log
## Modules
- [x] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/api/TaskLogWsServer.java [hunk 1]
```diff
@@ -41,39 +37,64 @@
...
 @ServerEndpoint("/api/v1/log/online/{taskId}")
-//@ServerEndpoint("${sw.controller.apiPrefix}/log/online/{taskId}")
 public class TaskLogWsServer {
 
-    private static final ConcurrentHashMap<String, TaskLogWsServer> sockets = new ConcurrentHashMap<>();
+    private static final ExecutorService executorService = Executors.newCachedThreadPool();
 
     private static IdConvertor idConvertor;
 
+    private static CancellableTaskLogK8sCollectorFactory logCollectorFactory;
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/api/TaskLogWsServer.java:46]: `static` is not needed

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/api/TaskLogWsServer.java:46]: I want this factory shared between the TaskLogWsServer instances.

Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/api/TaskLogWsServer.java:46]: `CancellableTaskLogK8sCollectorFactory` is a singleton bean held by Spring. And you have `setLogCollectorFactory` by `Autowired`. So it is shared even `static` is removed.

**Review by anda-ren — DISMISSED** [review]
lgtm

**Review by waynelwz — APPROVED** [review]
LGTM

**Review by anda-ren — APPROVED** [review]
lgtm

**Pull request merged**